R/createPatientSet.R

```diff
@@ -75,27 +75,27 @@ createPatientSet <- function(study.name, patientset.constraints, returnXMLquery
 
 .checkPatientSetConstraints <- function(patientsetConstraints){
     #test if it is expression and not a string. If string: try to parse
-    if(is.character(patientsetConstraints)){
-        if(length(patientsetConstraints) > 1){
-            stop("Incorrect input for patient set constraints. Found multiple strings for defining the patient set constraints. 
-					 The patient set constraints should be supplied in one single expression (or string).")}
-        
-        try({patientsetConstraintsParsed <- parse(text = patientsetConstraints)[[1]]
-             if(length(patientsetConstraintsParsed) == 1){
-                 if(is.character(patientsetConstraintsParsed)){ #e.g. happens if input string is "\"age\""
-                     patientsetConstraints <- patientsetConstraintsParsed
-                 }
-             }
-             if(length(patientsetConstraintsParsed) > 1){
-                 message(paste("\nDetecting a string as input for patient set constraints - expected is an expression,",  
-                               "such as: \"age\" > 65.",
-                               "\nWill attempt to parse the constraints out of the string, converting it",
-                               "into an expression..."))
-                 patientsetConstraints <- patientsetConstraintsParsed
-             }
-        }, silent = T
-        )
+    if(!is.character(patientsetConstraints)) {
+    	return(patientsetConstraints)
     }
+    if(length(patientsetConstraints) > 1){
+        stop("Incorrect input for patient set constraints. Found multiple strings for defining the patient set constraints. 
+				 The patient set constraints should be supplied in one single expression (or string).")}
+    
+	# TODO: is deze try nodig?
+    try({patientsetConstraintsParsed <- parse(text = patientsetConstraints)[[1]]
```

Comment:
I think you have been working on an older version of the code. See commit of 7 sept where I changed some code here: it catches the error message if parsing fails and gives a more comprehensible error message instead